test/lib/ufe/testConnections.py

```diff
@@ -421,5 +423,99 @@ def testConnectionsHandler(self):
+
+        materialItem = rootHier.children()[-1]
+        materialAttrs = ufe.Attributes.attributes(materialItem)
+        contextOps = ufe.ContextOps.contextOps(materialItem)
+
+        cmd = contextOps.doOpCmd(['Add New Prim', 'Shader'])
+        ufeCmd.execute(cmd)
+
+        materialHier = ufe.Hierarchy.hierarchy(materialItem)
+        materialPrim = usdUtils.getPrimFromSceneItem(materialItem)
+
+
+        shaderItem = materialHier.children()[0]
+        shaderAttrs = ufe.Attributes.attributes(shaderItem)
+        shaderPrim = usdUtils.getPrimFromSceneItem(shaderItem)
+
+        shaderAttr = shaderAttrs.attribute("info:id")
+        shaderAttr.set("ND_standard_surface_surfaceshader")
+        #
+        #
+        # Then switch to connection code to connect the shader. Since we never created the
+        # attributes, we expect the connection code to create them.
+        #
+        #
+        shaderOutput = shaderAttrs.attribute("outputs:out")
+        materialOutput = materialAttrs.attribute("outputs:surface")
+
+        connectionHandler = ufe.RunTimeMgr.instance().connectionHandler(materialItem.runTimeId())
+
+        # The attributes are not created yet:
+        self.assertEqual(len(materialPrim.GetAuthoredProperties()), 0)
+        self.assertEqual(len(shaderPrim.GetAuthoredProperties()), 1)
+        self.assertIn("info:id", [i.GetName() for i in shaderPrim.GetAuthoredProperties()])
+
+        connectionHandler.connect(shaderOutput, materialOutput)
+
+        connections = connectionHandler.sourceConnections(materialItem)
+        self.assertIsNotNone(connections)
+        conns = connections.allConnections()
+        self.assertEqual(len(conns), 1)
+
+        # The attributes directly used by the connection got created:
+        self.assertEqual(len(shaderPrim.GetAuthoredProperties()), 2)
+        self.assertIn("info:id", [i.GetName() for i in shaderPrim.GetAuthoredProperties()])
+        self.assertIn("outputs:out", [i.GetName() for i in shaderPrim.GetAuthoredProperties()])
+        # The connection of a MaterialX shader to the material got redirected to the proper render context.
+        self.assertEqual(len(materialPrim.GetAuthoredProperties()), 1)
+        self.assertEqual(materialPrim.GetAuthoredProperties()[0].GetName(), "outputs:mtlx:surface")
+
+        materialOutput = materialAttrs.attribute("outputs:mtlx:surface")
+        connectionHandler.disconnect(shaderOutput, materialOutput)
+
+        connections = connectionHandler.sourceConnections(materialItem)
+        self.assertIsNotNone(connections)
+        conns = connections.allConnections()
+        self.assertEqual(len(conns), 0)
+
+        # Not redirected since already on outputs:mtlx:surface:
+        connectionHandler.connect(shaderOutput, materialOutput)
+
+        connections = connectionHandler.sourceConnections(materialItem)
+        self.assertIsNotNone(connections)
+        conns = connections.allConnections()
+        self.assertEqual(len(conns), 1)
+
```

Comment:
The test can now create a connection i.e. it should fail because the connection already exists.
Note that the same check can be done with the delete connection.

Reply:
Same issue, disconnect return void/None.